sherpa/astro/io/pyfits_backend.py

```diff
@@ -119,43 +118,22 @@ def _get_meta_data(hdu):
 
     meta = {}
     for key, val in hdu.header.items():
-        # empty numpy strings are not recognized by load pickle!
-        if isinstance(val, numpy.str_) and val == '':
-            val = ''
-
         meta[key] = val
 
     return meta
 
 
-# Note: it is not really WCS specific, but leave the name alone for now.
-def _get_wcs_key(hdu, key0, key1):
-    """Return the pair of keyword values as an array of values of
-    the requested datatype. If either key is missing then return
-    ().
-    """
-
-    val1 = _try_key(hdu, key0)
-    if val1 is None:
-        return ()
-
-    val2 = _try_key(hdu, key1)
-    if val2 is None:
-        return ()
-
-    return numpy.array([val1, val2], dtype=SherpaFloat)
-
-
-def _add_keyword(hdrlist, name, val):
+def _add_keyword(hdrlist: fits.Header,
+                 name: str,
+                 val: Any) -> None:
     """Add the name,val pair to hdulist."""
-    name = str(name).upper()
-    if name in ['', 'COMMENT', 'HISTORY']:
-        # The values are assumed to be an array of strings
-        for v in val:
-            hdrlist.append(fits.Card(name, v))
 
-    else:
-        hdrlist.append(fits.Card(name, val))
+    # This used to special case name = { "", "COMMENT", "HEADER" } as
```

Comment:
```suggestion
    # This was used to special case name = { "", "COMMENT", "HEADER" } as
```
Just to make super clear that this is a comment on past code.
Or, just remove.